src/subcommands/server_test.go

```diff
@@ -0,0 +1,162 @@
+package subcommands
+
+import (
+	"context"
+	"os"
+	"path/filepath"
+	"testing"
+	"time"
+
+	pb "accretional.com/semantifly/proto/accretional.com/semantifly/proto"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+)
+
+const testDir = "./test_semantifly"
+
+func setupTestEnvironment(t *testing.T) func() {
+	err := os.MkdirAll(testDir, 0755)
+	if err != nil {
+		t.Fatalf("Failed to create test directory: %v", err)
+	}
+
+	testFiles := []string{"test_file1.txt", "test_file2.txt"}
+	for _, file := range testFiles {
+		filePath := filepath.Join(testDir, file)
+		err := os.WriteFile(filePath, []byte("This is a test file for Semantifly"), 0644)
+		if err != nil {
+			t.Fatalf("Failed to create test file %s: %v", file, err)
+		}
+	}
+
+	return func() {
+		// Clean up test directory
+		os.RemoveAll(testDir)
+	}
+}
+
+func setupServerAndClient(t *testing.T) (pb.SemantiflyClient, func()) {
+	startTestServer()
+
+	// Set up the client
+	opts := []grpc.DialOption{
+		grpc.WithTransportCredentials(insecure.NewCredentials()),
+	}
+
+	conn, err := grpc.NewClient("localhost:50051", opts...)
+	if err != nil {
+		t.Fatalf("Failed to create client: %v", err)
+	}
+
+	client := pb.NewSemantiflyClient(conn)
+
+	cleanup := func() {
+		conn.Close()
+	}
+
+	return client, cleanup
+}
+
+func startTestServer() {
+	go executeStartServer([]string{"--index-path", testDir})
+
+	// Allow some time for the server to start
+	time.Sleep(100 * time.Millisecond)
+}
+
+func TestServerCommands(t *testing.T) {
+	// Start server
+	cleanupTestEnv := setupTestEnvironment(t)
+	defer cleanupTestEnv()
+
+	client, cleanupClient := setupServerAndClient(t)
+	defer cleanupClient()
+
+	// Add
+	addCtx, addCancel := context.WithTimeout(context.Background(), time.Second)
+
+	testFileData1 := &pb.ContentMetadata{
+		DataType:   0,
+		SourceType: 0,
+		URI:        filepath.Join(testDir, "test_file1.txt"),
+	}
+
+	addArgs := &pb.AddRequest{
+		AddedMetadata: testFileData1,
+		MakeCopy:  true,
+	}
+
+	_, err := client.Add(addCtx, addArgs)
+	addCancel()
+	if err != nil {
+		t.Fatalf("Failed to add test file: %v", err)
+	}
+
+	// Get
+	getCtx, getCancel := context.WithTimeout(context.Background(), time.Second)
+	defer getCancel()
+
+	getReq := &pb.GetRequest{
+		Name: filepath.Join(testDir, "test_file1.txt"),
+	}
+
+	getResp, err := client.Get(getCtx, getReq)
+	if err != nil {
+		t.Fatalf("Get failed: %v", err)
+	}
+	if *getResp.Content == "" {
```

Comment:
verify that the actual contents match with what you set them as instead of this

Reply:
done